pkg/api/secretbootstrap/secretboostrap.go

```diff
@@ -79,6 +125,68 @@ func SaveConfigToFile(file string, config *Config) error {
 	return os.WriteFile(file, bytes, 0644)
 }
 
+// configForMarshal is used to avoid infinite recursion in MarshalYAML
+type configForMarshal struct {
+	VaultDPTPPrefix           string              `json:"vault_dptp_prefix,omitempty"`
+	ClusterGroups             map[string][]string `json:"cluster_groups,omitempty"`
+	Secrets                   []SecretConfig      `json:"secret_configs"`
+	UserSecretsTargetClusters []string            `json:"user_secrets_target_clusters,omitempty"`
+}
+
+// MarshalYAML ensures deterministic ordering of map keys and slices
+func (c *Config) MarshalYAML() (interface{}, error) {
```

Comment:
We could order them in the same place we load them. I don't think there is a need to wrap this and override the marshal method.